### ISSUE TITLE
Support SolidStart

### ISSUE THREAD

**Issue by paperclover** [oven-sh/bun#3736]
Now that Vite (#250) is working, it's now time to start getting all of the vite-powered frameworks working.

The current blockers for SolidStart:
- [ ] Dependency on node's --experiemntal-vm-modules flag
- [x] Event loop bug causing the process to exit prematurely while loading `es-module-loader` https://github.com/oven-sh/bun/pull/3741

**Comment by birkskyum** [comment]
**Repro:**

- Go to a new directory
- `bun x create-solid@latest`
- Select either option regarding template (bare is the simplest Hello World example)
- Select either option regarding SSR
- Select either option regarding TypeScript
- `bun i`
- `bun --bun run dev`

**Comment by Jarred-Sumner** [comment]
@birkskyum can you upgrade to canary and give it another try? `bun upgrade --canary`

**Comment by birkskyum** [comment]
@Jarred-Sumner , I experience the same with canary (hash: 3418feb2)

**Comment by birkskyum** [comment]
Update for canary (hash 3418feb2) 

### **solid-start: 0.3.4** (Currently active version)

Result of above repro:

`Cannot GET /` instead of showing the Hello World page

In more advanced cases (an app I work on) I get this result now using both Bun and Node adapter:

```
error when starting dev server:
NotImplementedError: node:http2 createSecureServer is not yet implemented in Bun. Track the status & thumbs up the issue: https://github.com/oven-sh/bun/issues/887
    at new NotImplementedError (node:http2:14:10)
    at throwNotImplemented (node:http2:2:68)
    at createSecureServer (node:http2:25:22)
```

Blocked by
* #8823

**Comment by birkskyum** [comment]
The repro for 0.2.27 appear to now work with bun canary 0.7.1.

**Comment by birkskyum** [comment]
Tried `movies` template of `bunx create-solid ./my-solid-app`, and it error with:

```sh
➜ bun --bun run dev -- --open
$ solid-start dev --open
 solid-start dev 
 version  0.3.2


error: Cannot find package "preact" from "file:///Users/admin/repos/my-solid-app/node_modules/local-pkg/index.mjs"

[bun] Warning: worker_threads.Worker option "execArgv" is not implemented.
```

- #4075 
- #4130

**Comment by quantuminformation** [comment]
yes pls

We've ported a SS project over to bun, but get this
```
[solid] file:///Users/nikos/WebstormProjects/coffeeconnect/node_modules/solid-start/node/fetch.js:73
[solid] export class NodeRequest extends Request {
[solid]                                  ^
```

**Comment by birkskyum** [comment]
Which adapter do you use?

**Comment by quantuminformation** [comment]
`googles  Node adapter`

**Comment by quantuminformation** [comment]
any docs for adaptors?

**Comment by birkskyum** [comment]
@quantuminformation , not sure I understand `
googles  Node adapter` - can you share your vite config?

**Comment by quantuminformation** [comment]
```js
import solid from 'solid-start/vite';
import { defineConfig } from 'vite';
import dotenv from 'dotenv';

export default defineConfig(() => {
  dotenv.config();
  return {
    plugins: [solid()],
    test: {
      deps: {
        registerNodeLoader: true,
      },
      environment: 'jsdom',
      globals: true,
      setupFiles: ['node_modules/@testing-library/jest-dom/extend-expect'],
      transformMode: { web: [/\.[jt]sx?$/] },
    },
    ssr: {
      external: ['@prisma/client'],
    },
  };
});
```

**Comment by marcusbuffett** [comment]
<img width="1333" alt="CleanShot 2023-09-13 at 14 57 47@2x" src="https://github.com/oven-sh/bun/assets/1834328/fdfe9857-c74d-4157-9b5c-4e6df2b6cfb8">

I get this when trying to run just a bare Solid Start project, with Bun 1.0.1. 

It seems to launch the server, but because of the Streams error, it doesn't actually serve on localhost:3000

**Comment by birkskyum** [comment]
@marcusbuffett, I believe it's this ticket - it's not implemented:
- #4075 
- #4130

**Comment by marcusbuffett** [comment]
@birkskyum I believe the worker thing isn't the main problem, it's the `globalThis` `TypeError` thing that's really breaking things – if I comment out that object assign, it serves pages

**Comment by quantuminformation** [comment]
Updated all deps:

![image](https://github.com/oven-sh/bun/assets/216566/a4b21672-6890-4b8a-a8aa-73ae9cdbd1cd)

same error
![image](https://github.com/oven-sh/bun/assets/216566/bd958a24-b1bd-45ac-bf03-79c1c32a1790)

**Comment by quantuminformation** [comment]
hmm I get the same error with `pnpm solid-start dev` prob not bun then

```
ReferenceError: Request is not defined
    at file:///Users/nikos/WebstormProjects/coffeeconnect/node_modules/.pnpm/solid-start@0.3.6_@solidjs+meta@0.28.6_@solidjs+router@0.8.3_solid-js@1.7.12_solid-start-node@0.3.6_vite@4.4.11/node_modules/solid-start/node/fetch.js:73:34
```

**Comment by quantuminformation** [comment]
my issue was fixed moving to node 18 from 16

**Comment by birkskyum** [comment]
Tried to run an [example](https://github.com/solidjs/solid-start/tree/vinxi/examples/bare) of the upcoming solid-start release. It's still WIP, so I expected bugs, but interestingly this example does work fine in node, but breaks with bun:


```zig
solid-start/examples/bare on  vinxi [$!?] via ⬢ v18.19.0 took 4.1s 
➜ bun --bun run dev
$ vinxi dev
36 |     );
37 | }
38 | 
39 | export function defineConfig(baseConfig = {}) {
40 |   let { plugins = [], ssr = true, start = {}, ...userConfig } = baseConfig;
41 |   const extensions = [...DEFAULT_EXTENSIONS, ...(start.extensions || [])];
                            ^
ReferenceError: Cannot access uninitialized variable.
      at defineConfig (/Users/admin/repos/solid-start/packages/start/config/index.js:41:25)
      at evalModule (/Users/admin/repos/solid-start/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/dist/jiti.js:1:256441)
      at jiti (/Users/admin/repos/solid-start/node_modules/.pnpm/jiti@1.21.0/node_modules/jiti/dist/jiti.js:1:254369)
      at /Users/admin/repos/solid-start/node_modules/.pnpm/c12@1.5.1/node_modules/c12/dist/index.mjs:288:15
      at resolveConfig (/Users/admin/repos/solid-start/node_modules/.pnpm/c12@1.5.1/node_modules/c12/dist/index.mjs:234:29)
      at /Users/admin/repos/solid-start/node_modules/.pnpm/c12@1.5.1/node_modules/c12/dist/index.mjs:116:39
      at loadConfig (/Users/admin/repos/solid-start/node_modules/.pnpm/c12@1.5.1/node_modules/c12/dist/index.mjs:86:26)
      at /Users/admin/repos/solid-start/node_modules/.pnpm/vinxi@0.0.48_debug@4.3.4_rollup@3.28.1/node_modules/vinxi/lib/load-app.js:35:28
      at processTicksAndRejections (:61:76)

vinxi hook app:config-resolved                                                              3:10:47 AM
vinxi hook app:created 
```

It claims that `DEFAULT_EXTENSIONS` is uninitialized, but in the top of [./packages/start/config/index.js](https://github.com/solidjs/solid-start/blob/vinxi/packages/start/config/index.js) where the error happen, it's defined as:

`const DEFAULT_EXTENSIONS = ["js", "jsx", "ts", "tsx"];`

**Comment by birkskyum** [comment]
There was a big new release on the solid-start side (@solidjs/start 0.4), but the error: `ReferenceError: Cannot access uninitialized variable.` was the same, and the initially presented reproduction instructions are still relevant as they now use this new version.

**Comment by itsyoboieltr** [comment]
As the latest version, it seems to be that `ReferenceError: Cannot access uninitialized variable.` is no longer an issue. The new error message I am getting is `zlib.brotliCompress is not implemented`. Could it be that this is the last missing piece of the puzzle to make this work?

Probably connected issue: https://github.com/oven-sh/bun/issues/267

**Comment by birkskyum** [comment]
The only issue I still face is the [server part of http2 missing](https://github.com/oven-sh/bun/issues/887) when https is turned on in localhost (which is quite common). Otherwise everything runs - do any of you still experience issues with latest versions of start (>=  0.4.3) and vinxi (>= 0.1.1)?

**Comment by willpuckett** [comment]
New versions seem to have fixed things here...

**Comment by itsyoboieltr** [comment]
Body parsing does not seem to work in solidstart production builds with bun (works with node): not sure if it is a bun or solidstart bug. Example code:
```ts
export const POST = async (event: APIEvent) => {
  console.log(await event.request.json());
  return new Response('ok');
};
```
The request hangs forever, instead of resolving.
cc: @birkskyum

**Comment by birkskyum** [comment]
If it works in node, it's to be considered a bug in bun - rule of thumb

**Comment by itsyoboieltr** [comment]
> If it works in node, it's to be considered a bug in bun - rule of thumb

Unless, they are doing some if-else thing specifically to handle bun, which has a bug :) Since nitro seems to have a specific bun adapter and preset, I would assume those have a bug somewhere, since it is not node code that is running.

**Comment by birkskyum** [comment]
Good point :)

**Comment by itsyoboieltr** [comment]
I am suspecting it is a [h3](https://github.com/unjs/h3) bug (or some h3 bun compatibility issue) since SolidStart uses h3 for api calls.
Reproduction of the issue with Bun: [solidstart-post-repro](https://github.com/itsyoboieltr/solidstart-post-repro)
Issue opened for Vinxi (that makes builds for SolidStart): https://github.com/nksaraf/vinxi/issues/95

**Comment by itsyoboieltr** [comment]
now post requests also work! It was fixed on their end in: https://github.com/solidjs/solid-start/pull/1255?fbclid=IwAR3uo6IrPCogXb-qZPUrLjiKc-WrCj_WFjKEVgJ617j9gX4SZihmNjlQbmE
Now everything seems to be working with SolidStart (dev, build, prod), so I guess the issue can finally be closed.

**Comment by birkskyum** [comment]
The [http2 server part](https://github.com/oven-sh/bun/issues/8823) is blocking for a lot of work with solid start apps as well as many other frameworks. It's not specific to this issue, but closing this before people confidently can use solid start does appear a bit too fast. I'd support closing it when it lands though.

**Comment by birkskyum** [comment]
Tested solid start 0.6.0 which is just out ([release notes](https://github.com/solidjs/solid-start/releases/tag/v0.6.0)).

Bun also breaks for single flight mutations:

```zig
➜ bun create solid
```

select solid-start, with typescript, **notes** example

```zig
➜ bun install
➜ bun --bun run dev
$ vinxi dev
vinxi v0.3.4
vinxi starting dev server

  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

1:01:59 AM [vite] ✨ new dependencies optimized: marked
1:01:59 AM [vite] ✨ optimized dependencies changed. reloading
```

Try to create a new note and save it - the server error is:

```zig
[h3] [unhandled] 1 | export default "native";
    ^
error: Unexpected end of JSON input
      at json (native:1:1)

1 | export default "native";
    ^
SyntaxError: Unexpected end of JSON input
      at json (native:1:1)
```

**Comment by birkskyum** [comment]
With the byonm flag, SolidStart is also stuck at
- #23043

**Comment by birkskyum** [comment]
This works now 🎉

**Comment by birkskyum** [comment]
Hit some error with missing crypto Diffie–Hellman

**Comment by paulo-assoc** [comment]
With bun v1.1.3 (tested on Windows 11) I get this error when doing a "bun run --bun dev" on a SolidStart (v1.0 RC - Clown Town) project:

Segmentation fault at address 0x20
???:?:?: 0x7ff7d1b2d621 in ??? (node.exe)
???:?:?: 0x7ff7d387d8b9 in ??? (node.exe)
???:?:?: 0x7ff7d212cd1f in ??? (node.exe)
???:?:?: 0x7ff7d21356a8 in ??? (node.exe)
???:?:?: 0x7ff7d208332e in ??? (node.exe)
???:?:?: 0x7ff7d20b6e0d in ??? (node.exe)
???:?:?: 0x7ff7d3802878 in ??? (node.exe)
???:?:?: 0x7ff7d1cb3cb7 in ??? (node.exe)
???:?:?: 0x7ff7d1d83b88 in ??? (node.exe)
???:?:?: 0x7ff7d37ce5da in ??? (node.exe)
???:?:?: 0x7ff7d16820c9 in ??? (node.exe)
???:?:?: 0x7ffd8bd8257c in ??? (KERNEL32.DLL)
???:?:?: 0x7ffd8d50aa47 in ??? (ntdll.dll)

**Comment by paulo-assoc** [comment]
And with the latest bun canary I get:

[h3] [unhandled] 1 | export default "native";
    ^
error: Unexpected end of JSON input
      at json (native:1:1)

1 | export default "native";
    ^
SyntaxError: Unexpected end of JSON input
      at json (native:1:1)

The same error as @birkskyum, but still happening in the latest bun canary as of April 10.

**Comment by paulo-assoc** [comment]
@paperdave @birkskyum You're our only hope!

**Comment by birkskyum** [comment]
Since the solid-start notes example runs fine with both Node `pnpm run dev` and Deno `DENO_FUTURE=1 deno task dev` this bug is highly likely to be found in Bun, and not in SolidStart, so at this point I'll leave it for the Bun team to debug further.

**Comment by paulo-assoc** [comment]
Thanks, @birkskyum! You're right. Jared said if something runs in node but not in bun, then it's a bug in bun.

**Comment by birkskyum** [comment]
One thing I did notice about this error is that it's only present in `bun --bun run dev`, but when making a build, and running it with `bun --bun run .output/server/index.mjs`, then the notes example works.

**Comment by paulo-assoc** [comment]
Very interesting find...

**Comment by birkskyum** [comment]
Wonder if related

https://twitter.com/jarredsumner/status/1782693859975373187

**Comment by paulo-assoc** [comment]
After upgrading to bun v1.1.5, solid-start notes sample app just hangs when the first http request comes in (bun run --bun). No errors in the console this time.

![Screenshot 2024-04-26 113050](https://github.com/oven-sh/bun/assets/163450497/2e801808-5ab4-4ca3-85a8-6fdc4794b4fa)

**Comment by birkskyum** [comment]
I have a weird bug with bun not being able to perform the `vinxi build` if the code use some files from the public folder, as those can not be found in the build step. 


```sh

$ bunx --bun vinxi build

⚙  Preparing app for bun...

 ERROR  { [ENOENT: No such file or directory]                                                                               3:09:45 PM
  code: 'ENOENT',
  path:
   '[absolute path]/public/.../scene.bin',
  syscall: 'copyfile',
  errno: -2 } 



 ERROR  No such file or directory                                                                                           3:09:45 PM

 ELIFECYCLE  Command failed with exit code 1.
 ```

**Comment by paulo-assoc** [comment]
Every release of bun fails differently when testing with the same solid-start notes sample app. After upgrading to bun v1.1.7, the server side now throws an http 500 error when trying to save a new note. No errors in the console this time. Ran on Windows 11.

![image](https://github.com/oven-sh/bun/assets/163450497/215ff7a5-3fc0-42c4-b6a1-6aaf3dffd576)

**Comment by paulo-assoc** [comment]
Solid-start notes sample app still fails with Bun v1.1.8 with this output in the console (when trying to save a new note):

bun run --bun dev
$ vinxi dev --port 3004
vinxi v0.3.11
vinxi starting dev server

  ➜ Local:    http://localhost:3004/
  ➜ Network:  use --host to expose

[h3] [unhandled] 1 | export default "native";
    ^
error: Unexpected end of JSON input
      at json (native:1:1)

1 | export default "native";
    ^
SyntaxError: Unexpected end of JSON input
      at json (native:1:1)

**Comment by birkskyum** [comment]
Issues with asset loading of files in public folder.

I experience that I can make a clean `basic` example, dump a few files in the public folder, like .glb 3d models, and then the `bun --bun run build` command will break with:

```sh
 ERROR  { [ENOENT: No such file or directory]                                                                                                            9:06:21 AM
  code: 'ENOENT',
  path:
   '/Users/admin/repos/bun-kitchensink/solid/wefwef/public/3dmodels/giantherb/giantherb.glb',
  syscall: 'copyfile',
  errno: -2 } 



 ERROR  No such file or directory                                                                                                                        9:06:21 AM

error: script "build" exited with code 1
```

It's worth noting that it's not a requirement to use these files anywhere in the code for this to break

**Comment by birkskyum** [comment]
Seems like this is the last only piece missing now for initial support:


```sh

➜ bun run --bun dev
$ vinxi dev
vinxi v0.3.11
vinxi starting dev server

  ➜ Local:    http://localhost:3000/
  ➜ Network:  use --host to expose

12:46:32 AM [vite] ✨ new dependencies optimized: marked
12:46:32 AM [vite] ✨ optimized dependencies changed. reloading
[h3] [unhandled] 1 | export default "native";
    ^
error: Unexpected end of JSON input
      at json (native:1:1)

1 | export default "native";
    ^
SyntaxError: Unexpected end of JSON input
      at json (native:1:1)
```

**Comment by posterhoudt** [comment]
Thanks, @birkskyum. Perhaps you can create a new issue with a more specific than "Support SolidStart". This worked great with #11260.

**Comment by birkskyum** [comment]
I'll try to see if I can get more out of the debug build of the bun binary

**Comment by birkskyum** [comment]
The bun-debug outputs this when saving a new note:

```sh
[SYS] send(6, 13) = 13
[FileSink] onWrite(13, src.io.PipeWriter.WriteStatus.drained)
[Loop] sub 13 - 0 = 13
[FileReader] onPull(16384) = pending
[Loop] sub 13 - 1 = 12
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/public/cross.svg) = 43
[alloc] new() = src.bun.js.webcore.blob.Blob.Store@169404080
[alloc] new() = src.bun.js.webcore.blob.Blob@169504080
[alloc] new() = src.bun.js.webcore.streams.ReadableStreamSource(src.bun.js.webcore.streams.FileReader,"File",(function 'onStart'),(function 'onPull'),(function 'onCancel'),(function 'deinit'),(function 'setRefOrUnref'),(function 'drain'))@162b04210
[SYS] dup(43) = 44
[SYS] fstat(44[/Users/admin/repos/bun-kitchensink/nodtes/public/cross.svg]) = 0
[alloc] new() = posix_event_loop.Closer@169604090
[FileReader] onReadChunk() = 207 (eof)
[FileReader] onReaderDone()
ASSERTION FAILED: microtaskArgs[0].isCallable()
/Users/admin/repos/bun/src/bun.js/bindings/bindings.cpp(5185) : void JSC__JSGlobalObject__queueMicrotaskJob(JSC__JSGlobalObject *, JSC__JSValue, JSC__JSValue, JSC__JSValue)
error: script "dev" was terminated by signal SIGTRAP (Trace or breakpoint trap)
zsh: trace trap  bun-debug --bun dev
```

**Comment by posterhoudt** [comment]
@birkskyum What build of Bun did you test with?

**Comment by birkskyum** [comment]
Main branch, bun-debug, and I just build it with this guide https://bun.sh/docs/project/contributing

**Comment by birkskyum** [comment]
All those "EAGAIN" doesn't look that healthy either, and that's even before clicking the save button.

**Comment by birkskyum** [comment]
Latest commit on main looks like this:

```sh
Latest commit on main looks like this:

```sh
[Subprocess] updateHasPendingActivity() true - true
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/routing.js) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/routing.js) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/routing.js])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/utils.js) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/utils.js) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/utils.js])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/routing.js) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/routing.js) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/@solidjs/router/dist/routing.js])
[napi] napi_get_value_bool
[napi] napi_create_promise
[napi] napi_create_string_utf8: napi_rs_async_work
[napi] napi_create_async_work
[napi] napi_queue_async_work
[Loop] ref 1000 + 1 = 1001
[napi] napi_get_value_bool
[napi] napi_create_promise
[napi] napi_create_string_utf8: napi_rs_async_work
[napi] napi_create_async_work
[napi] napi_queue_async_work
[Loop] ref 1001 + 1 = 1002
[FileReader] onPull(16384) = pending
[napi] napi_resolve_deferred
[napi] napi_delete_async_work
[Loop] sub 10 - 1 = 9
[napi] napi_resolve_deferred
[napi] napi_delete_async_work
[Loop] sub 9 - 1 = 8
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/dist/server.js) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/dist/server.js) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/dist/server.js])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/dist/package.json) = -1
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/package.json) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/package.json) = 38
[SYS] fstat(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/package.json]) = 0
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/package.json], 1331) = 1315 (0.011ms)
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/package.json], 16) = 0 (0.002ms)
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/solid-js/store/package.json])
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json])
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json) = 38
[SYS] fstat(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json]) = 0
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json], 158102) = 158086 (0.016ms)
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json], 16) = 0 (0.002ms)
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/package.json])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/index.mjs) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/index.mjs) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/date-fns/index.mjs])
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json])
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json) = 38
[SYS] fstat(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json]) = 0
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json], 3581) = 3565 (0.003ms)
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json], 16) = 0 (0.002ms)
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/lib/marked.esm.js) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/lib/marked.esm.js) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/lib/marked.esm.js])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/lib/package.json) = -1
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json) = 38
[SYS] fstat(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json]) = 0
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json], 3581) = 3565 (0.003ms)
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json], 16) = 0 (0.001ms)
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/marked/package.json])
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json])
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json) = 38
[SYS] fstat(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json]) = 0
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json], 4056) = 4040 (0.003ms)
[SYS] read(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json], 16) = 0 (0.001ms)
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/package.json])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/dist/index.mjs) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/dist/index.mjs) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/dist/index.mjs])
[SYS] stat(/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/drivers/fs-lite.mjs) = 0
[SYS] openat(-2, /Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/drivers/fs-lite.mjs) = 38
[SYS] close(38[/Users/admin/repos/bun-kitchensink/nodtes/node_modules/unstorage/drivers/fs-lite.mjs])
[h3] [unhandled] 1 | export default "native";
    ^
error: Unexpected end of JSON input
      at json (native:1:1)
      at asyncFunctionResume (:1:11)
      at promiseReactionJobWithoutPromiseUnwrapAsyncContext (:1:11)
      at promiseReactionJob (:1:11)
      at native:1:1

1 | export default "native";
    ^
SyntaxError: Unexpected end of JSON input
      at json (native:1:1)
      at asyncFunctionResume (:1:11)
      at promiseReactionJobWithoutPromiseUnwrapAsyncContext (:1:11)
      at promiseReactionJob (:1:11)
      at native:1:1


```
```

**Comment by posterhoudt** [comment]
Nice sleuthing, @birkskyum! How much closer does this get us to a resolution?

**Comment by birkskyum** [comment]
Not so much, but at least it seems like things are changing, and because all PRs have tests associated with them hopefully in a good direction. I mainly add the stacktraces, because they are searchable, like in the case of the "EAGAIN" which might have given us an open ticket to an issue which might be causing some sort of crash even before we click the button and see the JSON error. My gist at this point is that it's node:fs related, and I see work being done on that which is why I wanted to see what changed today.

**Comment by posterhoudt** [comment]
I appreciate what you're doing, @birkskyum! Was your test today using the latest canary build?

**Comment by birkskyum** [comment]
Yes, it was using the latest commit from main (1.1.13-debug+60af98586) so that should be the same.

**Comment by BierDav** [comment]
Hi there! From the last messages I didn't really pickup the current status. Is it currently possible to start a simple Hello World app? I would really appreaciate solid start support!

**Comment by birkskyum** [comment]
@BierDav , your milage might vary here, but a simple Hello World should be possible. The `notes` starter example that is breaking use some of the more advanced features like single-flight mutations.

**Comment by BierDav** [comment]
@birkskyum, thanks for your fast answer! I wanted to ask how to get this working, because the ´bun create` doesn't currenty work. I have already created an issue for that https://github.com/solidjs/solid-start/issues/1535

**Comment by birkskyum** [comment]
Use pnpm/npm create solid, and then when the project is created, delete any pnpm/npm lockfile and run `bun install`.

**Comment by BierDav** [comment]
Ok thanks, that worked. But is it also possible to use bun as a `vinxi` replacement too?

**Comment by birkskyum** [comment]
@BierDav , not that isn't possible. Vinxi was inspired by this idea os "[Bun.App](https://bun.sh/blog/bun-bundler#sneak-peek-bun-app)", but i don't think that ever shipped.

**Comment by posterhoudt** [comment]
Not unexpected, but with Bun 1.1.13 still getting 'SyntaxError: Unexpected end of JSON input' when saving a new note in the SolidStart Notes sample app.
![image](https://github.com/oven-sh/bun/assets/163450497/eb98f2e4-34a3-4351-a980-782c23321710)

**Comment by birkskyum** [comment]
In summary, these are the current issues related to SolidStart:

- http/2 server support: https://github.com/oven-sh/bun/issues/8823
- FormData bug fixes: https://github.com/oven-sh/bun/issues/2644
- Single Flight server actions (Notes example): https://github.com/oven-sh/bun/issues/3736#issuecomment-2105046143
- Better AuthJS support:  https://github.com/oven-sh/bun/issues/12178

**Comment by paulo-assoc** [comment]
Thanks, @birkskyum! These need 'bug' and 'triage' labels.

**Comment by znycheporuk** [comment]
with https://github.com/oven-sh/bun/issues/2644 closed, Bun 1.1.27 works fine with SolidStart in my case

**Comment by nektro** [comment]
looks like the linked issues have now been resolved
closing for now but if any other issues are discovered please don't hesitate to open a new one :)
